### PR TITLE
Adding scaphandre integration

### PR DESCRIPTION
**Pull request by rahulkaukuntla**
### What does this PR do?

[A brief description of the change being made with this pull request.](https://datadoghq.atlassian.net/wiki/spaces/~5e139434eb2ae70ea1dfe9df/pages/3709469147/Power+Usage+Monitoring+Carbon+Footprint+Monitoring#scaphandre)

### Review checklist

- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)
- [ ] If this PR includes a log pipeline, please add a description describing the remappers and processors. 

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: .github/workflows/test-all.yml [hunk 1]
```diff
@@ -1113,6 +1113,25 @@ jobs:
       test-py3: ${{ inputs.test-py3 }}
       setup-env-vars: "${{ inputs.setup-env-vars }}"
     secrets: inherit
+  ja91188c:
+    uses: DataDog/integrations-core/.github/workflows/test-target.yml@master
```
[the remark this answers is not in the document]
Comment (rahulkaukuntla) [reply .github/workflows/test-all.yml:1117]: done.

File: scaphandre/README.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+### Events
+
+Scaphandre does not include any events.
+
+## Troubleshooting
+
+Need help? Contact [Datadog support][3].
+
+[1]: **LINK_TO_INTEGRATION_SITE**
```
Comment (neko-dd) [inline scaphandre/README.md:54]: This needs to be updated

File: scaphandre/README.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+## Setup
+
+### Installation
+
+To install the Scaphandre check on your host:
+
+
+1. Install the [developer toolkit]
+(https://docs.datadoghq.com/developers/integrations/python/)
+ on any machine.
```
Comment (neko-dd) [inline scaphandre/README.md:16]: Please make this a reference link and specify which developer toolkit it is in the link text

File: scaphandre/README.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+2. Run `ddev release build scaphandre` to build the package.
+
+3. [Download the Datadog Agent][2].
+
+4. Upload the build artifact to any host with an Agent and
+ run `datadog-agent integration install -w
+ path/to/scaphandre/dist/<ARTIFACT_NAME>.whl`.
```
Comment (neko-dd) [inline scaphandre/README.md:24]:
~~~suggestion
4. Upload the build artifact to any host with an Agent and run the following command:
    ```
    datadog-agent integration install -w path/to/scaphandre/dist/<ARTIFACT_NAME>.whl
    ```
~~~

File: scaphandre/README.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+### Configuration
+
+Edit the `scaphandre.d/conf.yaml` file in the `conf.d/` folder at the root of your [Agent's configuration directory][10]. See the [sample scaphandre.d/conf.yaml][11] for all available configuration options. For example, to secure Scaphandre command line tags and prevent sensitive data from being pulled into Datadog, you can use the `exclude_labels` config option.
```
Comment (neko-dd) [inline scaphandre/README.md:28]:
```suggestion
Edit the `scaphandre.d/conf.yaml` file in the `conf.d/` folder at the root of your [Agent's configuration directory][10]. See the [sample scaphandre.d/conf.yaml][11] for all available configuration options. For example, to secure Scaphandre command line tags and prevent sensitive data from being pulled into Datadog, use the `exclude_labels` config option.
```

File: scaphandre/README.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+[Restart the Agent][12] to start sending Scaphandre metrics to Datadog.
+
+### Validation
+
+Run the [Agent's status subcommand][13] and look for `scaphandre` under the Checks section.
```
Comment (neko-dd) [inline scaphandre/README.md:34]:
```suggestion
Run the [Agent's status subcommand][13] and look for `scaphandre` in the **Checks** section.
```

File: scaphandre/README.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
+# Agent Check: Scaphandre
```
Comment (neko-dd) [inline scaphandre/README.md:1]: I think we usually just call these by the service name:

```suggestion
# Scaphandre
```

File: scaphandre/README.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+## Overview
+
+This check monitors [Scaphandre][1].
```
Comment (neko-dd) [inline scaphandre/README.md:5]: Please add information about what Scaphandre is/does, and why a customer would want to use it.

File: scaphandre/README.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+### Installation
+
+To install the Scaphandre check on your host:
+
+
+1. Install the [developer toolkit]
+(https://docs.datadoghq.com/developers/integrations/python/)
+ on any machine.
+
+2. Run `ddev release build scaphandre` to build the package.
```
Comment (neko-dd) [inline scaphandre/README.md:18]:
~~~suggestion
2. Run the following command to build the package:
    ```
    ddev release build scaphandre
    ```
~~~

File: scaphandre/README.md [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+
+3. [Download the Datadog Agent][2].
+
+4. Upload the build artifact to any host with an Agent and
```
Comment (neko-dd) [inline scaphandre/README.md:22]: What is "the build artifact"? Can this be more specific?

File: scaphandre/README.md [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+
+Scaphandre does not include any service checks.
+
+### Events
+
+Scaphandre does not include any events.
+
+## Troubleshooting
+
+Need help? Contact [Datadog support][3].
```
Comment (neko-dd) [inline scaphandre/README.md:52]: This link reference doesn't seem to match the provided link below.

File: scaphandre/README.md [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+
+[1]: **LINK_TO_INTEGRATION_SITE**
+[2]: https://app.datadoghq.com/account/settings/agent/latest
+[3]: https://docs.datadoghq.com/agent/kubernetes/integrations/
+[4]: https://github.com/DataDog/integrations-extras/blob/master/scaphandre/datadog_checks/scaphandre/data/conf.yaml.example
+[5]: https://docs.datadoghq.com/agent/guide/agent-commands/#start-stop-and-restart-the-agent
+[6]: https://docs.datadoghq.com/agent/guide/agent-commands/#agent-status-and-information
+[7]: https://github.com/DataDog/integrations-extras/blob/master/scaphandre/metadata.csv
+[8]: https://github.com/DataDog/integrations-extras/blob/master/scaphandre/assets/service_checks.json
+[9]: https://docs.datadoghq.com/help/
```
Comment (neko-dd) [inline scaphandre/README.md:62]: It doesn't look like these links are used above.

File: scaphandre/assets/dashboards/scaphandre_overview.json [hunk 13]
```diff
@@ -0,0 +1,873 @@
...
+                "banner_img": "https://repository-images.githubusercontent.com/304646001/689da800-3958-11eb-8117-3efc5fe2a802",
+                "show_title": false,
+                "type": "group",
+                "layout_type": "ordered",
+                "widgets": [
+                    {
+                        "id": 5478983280988798,
+                        "definition": {
+                            "type": "note",
+                            "content": "**Scaphandre**\n\nThis dashboard provides an overview of your [Scaphandre](https://github.com/hubblo-org/scaphandre) metrics.\n\nView the power consumption of your hosts, as well as the power consumption per process to determine the most energy inefficient processes in your deployment",
```
Comment (neko-dd) [inline scaphandre/assets/dashboards/scaphandre_overview.json:18]:
```suggestion
                            "content": "**Scaphandre**\n\nThis dashboard provides an overview of your [Scaphandre](https://github.com/hubblo-org/scaphandre) metrics.\n\nView the power consumption of your hosts, as well as the power consumption per process, to determine the most energy inefficient processes in your deployment",
```

File: scaphandre/assets/dashboards/scaphandre_overview.json [hunk 14]
```diff
@@ -0,0 +1,873 @@
...
+                            "y": 0,
+                            "width": 3,
+                            "height": 3
+                        }
+                    },
+                    {
+                        "id": 6394363111995732,
+                        "definition": {
+                            "type": "note",
+                            "content": "**Useful Links**\n\n\n* [Scaphandre Github](https://github.com/hubblo-org/scaphandre)\n* [Scaphandre Documentation](https://hubblo-org.github.io/scaphandre-documentation/index.html)",
```
Comment (neko-dd) [inline scaphandre/assets/dashboards/scaphandre_overview.json:39]:
```suggestion
                            "content": "**Useful Links**\n\n\n* [Scaphandre on Github](https://github.com/hubblo-org/scaphandre)\n* [Scaphandre Documentation](https://hubblo-org.github.io/scaphandre-documentation/index.html)",
```

File: scaphandre/metadata.csv [hunk 15]
```diff
@@ -0,0 +1,30 @@
+metric_name,metric_type,interval,unit_name,per_unit_name,description,orientation,integration,short_name,curated_metric
+scaphandre.host.cpu.frequency,gauge,,,,,0,scaphandre,scaphandre host freq,
```
Comment (neko-dd) [inline scaphandre/metadata.csv:2]: This line is missing a description string.

Comment (rahulkaukuntla) [reply scaphandre/metadata.csv]: added

File: scaphandre/README.md (hunk 12 above)
Comment (rahulkaukuntla) [reply scaphandre/README.md]: Removed these links.

File: scaphandre/README.md (hunk 11 above)
Comment (rahulkaukuntla) [reply scaphandre/README.md]: Updated the link.

File: scaphandre/README.md (hunk 8 above)
Comment (rahulkaukuntla) [reply scaphandre/README.md]: Added a little description about scaphandre.

File: scaphandre/README.md (hunk 3 above)
Comment (rahulkaukuntla) [reply scaphandre/README.md]: Done--the actual toolkit depends on the user's platform (Mac vs Windows), so I added a little note about that too.

File: scaphandre/README.md (hunk 2 above)
Comment (rahulkaukuntla) [reply scaphandre/README.md]: Done.

File: scaphandre/README.md (hunk 4 above)
Comment (rahulkaukuntla) [reply scaphandre/README.md]: Done.

File: scaphandre/README.md (hunk 5 above)
Comment (rahulkaukuntla) [reply scaphandre/README.md]: Done.

File: scaphandre/README.md (hunk 7 above)
Comment (rahulkaukuntla) [reply scaphandre/README.md:1]: Done.

File: scaphandre/README.md (hunk 6 above)
Comment (rahulkaukuntla) [reply scaphandre/README.md]: Done.

File: scaphandre/README.md [hunk 16]
```diff
@@ -0,0 +1,65 @@
+# Scaphandre
+
+## Overview
+
+This check monitors [Scaphandre][1], a monitoring agent that uses RAPL and MsrRAPL via powercap to measure power usage of bare metal machines. The goal of the project is to permit to any company or individual to measure the power consumption of its tech services and get this data in a convenient form, sending it through any monitoring or data analysis toolchain.
```
Comment (neko-dd) [inline scaphandre/README.md:5]:
```suggestion
This check monitors [Scaphandre][1], a monitoring agent that uses RAPL and MsrRAPL through powercap to measure power usage of bare metal machines. The goal of the project is to permit to any company or individual to measure the power consumption of its tech services and get this data in a convenient form, sending it through any monitoring or data analysis toolchain.
```

**Review by neko-dd — APPROVED** [review]

**Review by HadhemiDD — APPROVED** [review]

File: scaphandre/CHANGELOG.md [hunk 17]
```diff
@@ -0,0 +1,8 @@
+# CHANGELOG - Scaphandre
+
+## 1.0.0 / 2024-05-16
```
Comment (iliakur) [inline scaphandre/CHANGELOG.md:3]: This should be managed by towncrier. Out of curiosity, which `ddev create` command did you run when starting to work on this integration?

**Review by iliakur — CHANGES_REQUESTED** [review]

**Review by iliakur — CHANGES_REQUESTED** [review]
plz switch the changelog to be managed by towncrier.

**Review by iliakur — COMMENTED** [review]
Oops, nevermind the changelog feedback :facepa

**Pull request merged**